projects/vdk-plugins/vdk-ipython/src/vdk_ipython.py

```diff
@@ -1,34 +1,84 @@
 # Copyright 2021-2023 VMware, Inc.
 # SPDX-License-Identifier: Apache-2.0
+import json
 import os
 import pathlib
 
 from IPython import get_ipython
 from IPython.core.magic_arguments import argument
 from IPython.core.magic_arguments import magic_arguments
 from IPython.core.magic_arguments import parse_argstring
-from vdk.internal.builtin_plugins.run.standalone_data_job import (
-    StandaloneDataJobFactory,
-)
+from vdk.internal.builtin_plugins.run import standalone_data_job
+
+
+class JobControl:
+    """
+    A class introducing an easier way to  work with IStandaloneDataJob instances
+    Arguments:
+            path: str
+                The source code of the data job that will be executed.
+            name: str
+                The name of the job.  If omitted will be inferred from the director name.
+            arguments: str
+                Allows for users to pass arguments to data job run.
+                Data Job arguments are also used for parameter substitution in queries, see execute_query docstring.
+            template: str
+                Template name, in case the data job is a template execution.
+    """
+
+    def __init__(
+        self,
+        path: str = None,
+        name: str = None,
+        arguments: str = None,
+        template: str = None,
+    ):
+        path = pathlib.Path(path) if path else pathlib.Path(os.getcwd())
+        job_args = json.loads(arguments) if arguments else None
+        self.job = standalone_data_job.StandaloneDataJob(
+            name=name,
+            template_name=template,
+            job_args=job_args,
+            data_job_directory=path,
+        )
+        self.job_input = None
+
+    def get_initialized_job_input(self):
+        """
+        Get initialised IJobInput object for the current job if present
+            :return:  an IJobInput object
+        """
+        if not self.job_input:
+            self.job_input = self.job.__enter__()
+        return self.job_input
+
+    # TODO: should automate calling finalize (if it is not called explicitly)
+    def finalise(self):
```

Comment:
```suggestion
    def finalize(self):
```

I think we stick mostly to American English spelling of words (e.g Initialize not Initialise)

Reply:
done